src/Fantomas/FormatConfig.fs

```diff
@@ -32,7 +43,18 @@ type FormatConfig =
         { IndentSpaceNum = 4
           PageWidth = 120
           SemicolonAtEndOfLine = false
-          SpaceBeforeArgument = true
+          SpaceBeforeUnitArgumentInUppercaseFunctionCall = false
+          SpaceBeforeUnitArgumentInLowercaseFunctionCall = false
+          SpaceBeforeParenthesesArgumentInUppercaseFunctionCall = false
+          SpaceBeforeParenthesesArgumentInLowercaseFunctionCall = true
+          SpaceBeforeUnitParameterInUppercaseFunctionDefinition = false
+          SpaceBeforeUnitParameterInLowercaseFunctionDefinition = false
+          SpaceBeforeParenthesesInUppercaseFunctionDefinition = false
+          SpaceBeforeParenthesesInLowercaseFunctionDefinition = true
+          SpaceBeforeUnitParameterInUppercaseClassConstructor = false
+          SpaceBeforeUnitParameterInLowercaseClassConstructor = false
+          SpaceBeforeParenthesesParameterInUppercaseClassConstructor = false
+          SpaceBeforeParenthesesParameterInLowercaseClassConstructor = false
```

Comment:
tbh I still think this explosion of config settings is a bit ludicrous, if I were you I would just have 3: SpaceBeforeFunctionCall, SpaceBeforeFunctionDefinition, SpaceBeforeClassConstructor

Reply:
All these settings are necessary because Fantomas now has a weird way of dealing with this.
In order to not break the existing behaviour, this fine-grained configuration is necessary.

Simply said if you format with the current version of Fantomas and later with a version containing this PR you want exactly the same output. Otherwise, this is a breaking change that will upset people.

Reply:
> In order to not break the existing behaviour, this fine-grained configuration is necessary.

I get that, but my 2cents assume this catch. My view on this is: the past behaviour was wrong, and there are not many cases out there that would be affected by it, so it should be fine correcting this inconsistency. But hey I'm no maintainer, your call to make thins unnecessarily complex because of some paranoid fear ;)